### PR TITLE
MCO-1103: Restore setup-envtest used in the verify test

### PR DESCRIPTION
**Pull request by dkhater-redhat**
**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -138,4 +141,8 @@ test-e2e-single-node: install-go-junit-report
 	set -o pipefail; go test -tags=$(GOTAGS) -failfast -timeout 120m -v$${WHAT:+ -run="$$WHAT"} ./test/e2e-single-node/ | ./hack/test-with-junit.sh $(@)
 
 bootstrap-e2e: install-go-junit-report install-setup-envtest
-	set -o pipefail; go test -tags=$(GOTAGS) -v$${WHAT:+ -run="$$WHAT"} ./test/e2e-bootstrap/ | ./hack/test-with-junit.sh $(@)
+	set -o pipefail; \
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --remote-bucket openshift-kubebuilder-tools --use-deprecated-gcs --bin-dir $(PROJECT_DIR)/bin)" \
+	echo "KUBEBUILDER_ASSETS set to $$KUBEBUILDER_ASSETS"; \
```
Comment (JoelSpeed) [inline Makefile:146]: I think this line is causing issues, you have `KUBEBUILDER_ASSETS` set here, but then on the next line (because of the semi-colon), you've broken the env var setting and go test is not getting the value set

File: Makefile [hunk 2]
```diff
@@ -138,4 +141,8 @@ test-e2e-single-node: install-go-junit-report
 	set -o pipefail; go test -tags=$(GOTAGS) -failfast -timeout 120m -v$${WHAT:+ -run="$$WHAT"} ./test/e2e-single-node/ | ./hack/test-with-junit.sh $(@)
 
 bootstrap-e2e: install-go-junit-report install-setup-envtest
```
Comment (JoelSpeed) [inline Makefile:143]: I'd suggest getting rid of this if you're going down the vendored route
```suggestion
bootstrap-e2e: install-go-junit-report
```

File: Makefile [hunk 3]
```diff
@@ -64,13 +65,16 @@ go-deps:
 	chmod +x ./vendor/k8s.io/code-generator/generate-groups.sh
 	chmod +x ./vendor/k8s.io/code-generator/generate-internal-groups.sh
 
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
+ENVTEST_K8S_VERSION = 1.29.1
```
Comment (JoelSpeed) [inline Makefile:69]: You probably want 1.30.1 now

File: pkg/daemon/daemon.go [hunk 4]
```diff
@@ -1328,7 +1328,7 @@ type pipeErrorHandler struct {
 	pipe chan error
 }
 
-func (e *pipeErrorHandler) handle(err error) {
+func (e *pipeErrorHandler) handle(_ context.Context, err error, _ string, _ ...interface{}) {
```
Comment (cheesesashimi) [inline pkg/daemon/daemon.go:1331]: question: Was this change needed here? I suspect this might be something that got caught between branches.

File: go.mod [hunk 5]
```diff
@@ -43,7 +43,7 @@ require (
 	google.golang.org/grpc v1.59.0
 	k8s.io/api v0.30.1
 	k8s.io/apiextensions-apiserver v0.30.1
-	k8s.io/apimachinery v0.30.1
+	k8s.io/apimachinery v0.31.0-alpha.1
```
Comment (cheesesashimi) [inline go.mod:46]: question: Do we want to bump this to a different version? Could we still add `setup-envtest` with `v0.30.1` of this library?

File: Makefile [hunk 6]
```diff
@@ -64,13 +65,16 @@ go-deps:
...
+ENVTEST_K8S_VERSION = 1.30.1
+ENVTEST = go run ${PROJECT_DIR}/vendor/sigs.k8s.io/controller-runtime/tools/setup-envtest
 SETUP_ENVTEST := $(shell command -v setup-envtest 2> /dev/null)
 install-setup-envtest:
 ifdef SETUP_ENVTEST
 	@echo "Found setup-envtest"
 else
 	@echo "Installing setup-envtest"
-	go install -mod= sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240315194348-5aaf1190f880
+	go install -mod= sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
```
Comment (cheesesashimi) [inline Makefile:77]: question: Do we need the `@latest` here? I _think_ that `go install` should default to the version that's in the `vendor/` dir, but I could be wrong. You may also have to lose the `-mod=` flag to make that work.

File: Makefile [hunk 7]
```diff
@@ -138,4 +142,8 @@ test-e2e-single-node: install-go-junit-report
 	set -o pipefail; go test -tags=$(GOTAGS) -failfast -timeout 120m -v$${WHAT:+ -run="$$WHAT"} ./test/e2e-single-node/ | ./hack/test-with-junit.sh $(@)
 
 bootstrap-e2e: install-go-junit-report install-setup-envtest
-	set -o pipefail; go test -tags=$(GOTAGS) -v$${WHAT:+ -run="$$WHAT"} ./test/e2e-bootstrap/ | ./hack/test-with-junit.sh $(@)
+	@echo "Setting up KUBEBUILDER_ASSETS"
```
Comment (cheesesashimi) [inline Makefile:145]: thought (non-blocking): I think this setup step may be better under the `install-setup-envtest` target, but if it's too difficult to make that work, then don't worry about it.

**Review by cheesesashimi — COMMENTED** [review]
Overall it looks good. I just have a couple of questions and comments.

File: pkg/daemon/daemon.go (hunk 4 above)
Comment (dkhater-redhat) [reply pkg/daemon/daemon.go:1331]: yeah it was :/ it was needed post bump. utilruntime.ErrorHandlers expects a value of a specific type, and the function (&pipeErrorHandler{errCh}).handle did not match the expected type

File: go.mod (hunk 5 above)
Comment (dkhater-redhat) [reply go.mod:46]: i didnt explicitly go get that version. it must've happened when i got setup-envtest. I can double check

Comment (dkhater-redhat) [reply go.mod:46]: ` go mod graph | grep k8s.io/apimachinery`
...
`sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240709080737-a0c9fd9d3f31 k8s.io/apimachinery@v0.31.0-alpha.1`

File: Makefile (hunk 6 above)
Comment (dkhater-redhat) [reply Makefile:77]: so the story for this PR wanted me to pull latest, but we can pull from go.mod if needed (probably is better).

File: Makefile (hunk 7 above)
Comment (dkhater-redhat) [reply Makefile:145]: thats what i was trying to do initially, but didnt get it to work. ill try a new approach but if i cant get it, i wont sweat it

Comment (dkhater-redhat) [reply Makefile:145]: resolving this- zack and i decided to stick with keeping it under the bootstrap-e2e target

**Comment by ptalgulk01** [comment]
Pre verified by running all MCO critical test cases in a IPI on AWS cluster which are all passed.
